pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ClientCnx.java

```diff
@@ -377,6 +378,7 @@ SocketAddress serverAddrees() {
         ctx.writeAndFlush(cmd).addListener(writeFuture -> {
             if (!writeFuture.isSuccess()) {
                 log.warn("{} Failed to send request to broker: {}", ctx.channel(), writeFuture.cause().getMessage());
+                pendingRequests.remove(requestId);
```

Comment:
no, actually we had a bug where if we failed to publish message on socket then we should remove from the queue.  line:341 removes from `lookup-PendingRequests` and this one removes from `pendingRequests`